spring-kafka/src/main/java/org/springframework/kafka/support/TopicPartitionInitialOffset.java

```diff
@@ -25,28 +25,64 @@
  * an initial offset for it. The initial offset can be:
  * <ul>
  * <li>{@code null} - do nothing;</li>
- * <li>positive (including {@code 0}) - seek to the absolute offset within the partition;
+ * <li>positive (including {@code 0}) - seek to EITHER the absolute offset within the
+ * partition or an offset relative to the current position for this consumer, depending
+ * on {@link #isRelativeToCurrent()}.
  * </li>
- * <li>negative - seek to the offset relative to the current last offset within the
- * partition: {@code consumer.seekToEnd() + initialOffset}.</li>
+ * <li>negative - seek to EITHER the offset relative to the current last offset within
+ * the partition: {@code consumer.seekToEnd() + initialOffset} OR the relative to the
+ * current offset for this consumer (if any), depending on
+ * {@link #isRelativeToCurrent()}.</li>
  * </ul>
  * Offsets are applied when the container is {@code start()}ed.
  *
  * @author Artem Bilan
+ * @author Gary Russell
  */
 public class TopicPartitionInitialOffset {
 
 	private final TopicPartition topicPartition;
 
 	private final Long initialOffset;
 
+	private final boolean relativeToCurrent;
```

Comment:
Yeah... That's good one argument. OK. Let's switch to your other PR, fix it merge and release :smile: 
Than I will have time to fight with `handleCancel`/`handleCancelOk` in SA :wink: